### PR TITLE
Added a Scheduler to run tasks at a given time

### PR DESCRIPTION
**Pull request by jcague**
This PR adds a new Scheduler to Workers, to be able to schedule tasks that we want to execute in the future. It's interesting to create periodic tasks, timeouts, etc.

### CODE REVIEW

File: erizo/src/erizo/thread/Scheduler.cpp [hunk 1]
```diff
@@ -0,0 +1,90 @@
+#include "thread/Scheduler.h"
+
+#include <assert.h>
+
+#include <boost/bind.hpp>
+#include <utility>
+
+
+Scheduler::Scheduler(int n_threads_servicing_queue)
+: n_threads_servicing_queue_(0), stop_requested_(false), stop_when_empty_(false) {
```
Comment (lodoyun) [inline erizo/src/erizo/thread/Scheduler.cpp]: shouldn't this be `Scheduler::Scheduler(int n_threads_servicing_queue)
 +: n_threads_servicing_queue_(n_threads_servicing_queue), stop_requested_(false), stop_when_empty_(false)`

File: erizo/src/erizo/thread/Scheduler.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    }
+  }
+  new_task_scheduled_.notify_all();
+  group_.join_all();
+}
+
+void Scheduler::schedule(Scheduler::Function f, std::chrono::system_clock::time_point t) {
+  {
+    std::unique_lock<std::mutex> lock(new_task_mutex_);
+    task_queue_.insert(std::make_pair(t, f));
```
Comment (lodoyun) [inline erizo/src/erizo/thread/Scheduler.cpp]: Considering including a comment or a reminder that the tasks are ordered by time_point in the multimap, to make it easier to contribute and avoid confusions

File: erizo/src/erizo/thread/Scheduler.cpp (hunk 1 above)
Comment (jcague) [inline erizo/src/erizo/thread/Scheduler.cpp]: well, internally we increase n_threads_servicing_queue_ from each thread. Either way is good IMO so if you feel strong about setting it in the constructor I can't change the code.

File: erizo/src/erizo/thread/Scheduler.cpp (hunk 2 above)
Comment (jcague) [inline erizo/src/erizo/thread/Scheduler.cpp]: done

File: erizo/src/erizo/thread/Scheduler.cpp (hunk 1 above)
Comment (jcague) [inline erizo/src/erizo/thread/Scheduler.cpp]: I finally apply your comment to avoid potential race conditions while running unit tests.

**Comment by jcague** [comment]
@lodoyun I've added  Unit Tests for the Scheduler.

**Pull request merged**